conftest.py

```diff
@@ -276,17 +296,22 @@ def serialize_datetimes(obj):
 
 
 def extract_metadata(resource):
-    return {
-        metadata_key: resource[metadata_key]
+    if isinstance(resource, (dict,)):
```

Comment:
Does this just copy the entire resource rather than iterating through `METADATA_KEYS` if `resource` is a dict?

Reply:
I'll research this one -- I vaguely recall adding this due to an error, but agree it needs a comment if it's valid.

Reply:
> Does this just copy the entire resource rather than iterating through `METADATA_KEYS` if `resource` is a dict?

No copying involved. Added some comments to clarify (I hope) in next commit.